phpunit/functional/UserTest.php

```diff
@@ -1012,6 +1012,113 @@ public function testGetFriendlyName($input, $rawname)
+                    'neither' => 'user4'
+                ]
+            ]
+        ];
+
+        $this->login();
+
+        // Create test users with different combinations of name/firstname
+        $user_data = [
+            // User with both firstname and realname
+            [
+                'name' => 'user1',
+                'realname' => 'Doe',
+                'firstname' => 'John',
+            ],
+            // User with only realname
+            [
+                'name' => 'user2',
+                'realname' => 'Smith',
+                'firstname' => '',
+            ],
+            // User with only firstname
+            [
+                'name' => 'user3',
+                'realname' => '',
+                'firstname' => 'Alice',
+            ],
+            // User with neither realname nor firstname
+            [
+                'name' => 'user4',
+                'realname' => '',
+                'firstname' => '',
+            ]
+        ];
+
+        // Create all test users
+        $user = new \User();
+        $user_ids = [];
+
+        foreach ($user_data as &$data) {
+            $id = (int)$user->add([
+                'name' => $data['name'],
+                'realname' => $data['realname'],
+                'firstname' => $data['firstname'],
+            ]);
+            $this->assertGreaterThan(0, $id);
+            $user_ids[] = $id;
+        }
+
+        // Test each name format configuration
+        foreach ($name_formats as $format => $format_data) {
+            // Save current config
+            $current_format = \Config::getConfigurationValues('core')['names_format'];
+
+            // Set the test format
+            \Config::setConfigurationValues('core', ['names_format' => $format]);
+
+            // Test the SQL query properly using the GLPI query API
+            $alias = 'test_name';
+            $field_expr = User::getFriendlyNameFields($alias);
+
+            // Use GLPI query API instead of direct SQL
+            $iterator = $DB->request([
+                'SELECT' => ['id', new \QueryExpression($field_expr)],
+                'FROM'   => 'glpi_users',
+                'WHERE'  => ['id' => $user_ids],
+                'ORDER'  => 'id'
+            ]);
+
+            $i = 0;
+            $expected_values = array_values($format_data['expected']);
+            foreach ($iterator as $row) {
+                // Verify we get the expected friendly name based on our test data
+                $this->assertEquals(
+                    $expected_values[$i],
+                    $row[$alias],
+                    "Failed with format {$format_data['label']} for user with name={$user_data[$i]['name']}, realname={$user_data[$i]['realname']}, firstname={$user_data[$i]['firstname']}"
+                );
+                $i++;
+            }
+
+            // Restore original config
+            \Config::setConfigurationValues('core', ['names_format' => $current_format]);
```

Comment:
I think this is already handled by the `resetGlobalsAndStaticValues()` method in `GLPITestCase`.